artemis-server/src/test/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManagerTest.java

```diff
@@ -145,4 +166,73 @@ public void testFixJournalFileSize() {
       Assert.assertEquals(4096, manager.fixJournalFileSize(4098, 4096));
       Assert.assertEquals(8192, manager.fixJournalFileSize(8192, 4096));
    }
+
+   @Test(timeout = 20_000)
```

Comment:
This test is hard coded to libaio.

Please, either make it runnable in other platforms, or add an Assume call to ignore the test in case of libaio:

 org.junit.Assume.assumeTrue("Test case needs AIO to run", AIOSequentialFileFactory.isSupported());


It's probably best if you move the new test to a new test class, and add this on the @BeforeClass:


```java

   @BeforeClass
   public static void hasAIO() {
      org.junit.Assume.assumeTrue("Test case needs AIO to run", AIOSequentialFileFactory.isSupported());
   }
```


An example of a class playing with Assume is AIOImportExportTest.


But you also have the option to make it runnable with NIO. But if not possible, please add the Assume call here.

Reply:
:+1: Thanks to have taken a look bud, please check if now is ok (I don't have a Mac/Win to test it!)